relay-server/src/endpoints/common.rs

```diff
@@ -303,17 +303,9 @@ fn queue_envelope(
             .map_err(BadStoreRequest::QueueFailed)?;
         state.project_cache().send(ValidateEnvelope::new(envelope));
     }
-
-    if managed_envelope.envelope().is_empty() {
-        // The envelope can be empty here if it contained only metrics items which were removed
-        // above. In this case, the envelope was accepted and needs no further queueing.
-        managed_envelope.accept();
-    } else {
-        relay_log::trace!("queueing envelope");
-        state
-            .project_cache()
-            .send(ValidateEnvelope::new(managed_envelope));
-    }
+    // The entire envelope is taken for a split above, and it's empty at this point, we can just
+    // accept it without additional checks.
+    managed_envelope.accept();
```

Comment:
Should we make `split_envelope` consume the managed envelope, so we don't have to handle its leftovers here?

Reply:
I might take this as followup, since right now this function also used in the place when we restore the envelopes from the spool, and there we have only boxed envelope. 
I could expose 2 function, once (which excepts the managed envelope) wraps another one (the current which jsut needs envelop) , it's that cleaner